src/arch/amdgpu/vega/insts/op_encodings.cc

```diff
@@ -1573,12 +1575,12 @@ namespace VegaISA
     Inst_FLAT::initOperandInfo()
     {
         // One of the flat subtypes should be specified via flags
-        assert(isFlat() ^ isFlatGlobal());
+        assert(isFlat() ^ isFlatGlobal() ^ isFlatScratch());
```

Comment:
Not relevant to this commit, but why is this an XOR (^) instead of an OR (|)?

Reply:
It must be exactly one and only one of those types, as that will determine which memory pipeline the request goes to and also how the address will be calculated